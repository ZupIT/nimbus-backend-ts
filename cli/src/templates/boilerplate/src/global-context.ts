import { getGlobalState } from '@zup-it/nimbus-backend-core'

export interface GlobalState {
  message?: string,
}

export const globalState = getGlobalState<GlobalState>()
