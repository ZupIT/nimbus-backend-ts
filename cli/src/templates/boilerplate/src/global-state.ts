import { getGlobalState } from '@zup-it/nimbus-backend-core'

export interface GlobalState {
  // here you should write every property in your global state. If you don't use a global state, you can safely
  // delete this file.
}

export const globalState = getGlobalState<GlobalState>()
