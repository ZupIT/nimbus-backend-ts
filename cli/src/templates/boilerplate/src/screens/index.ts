import { RouteMap } from '@zup-it/nimbus-backend-express'
import { Home } from './home'
import { Welcome } from './welcome'

export const routes: RouteMap = {
  '': Welcome,
  '/home': Home,
}
