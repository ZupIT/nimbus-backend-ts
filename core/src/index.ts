export { Component, WithState, WithChildren } from './model/component'
export { Expression, HttpMethod, DynamicExpression, DeepExpression } from './types'
export { RootState } from './model/state/root-state'
export { Actions, createAction } from './model/action'
export { State } from './model/state/types'
export { createStateNode } from './model/state/state-node'
export { createState } from './model/state/root-state'
export { serialize } from './serializer'
export { FC, ComponentProps } from './jsx/types'
export { isDynamicExpression, setupHotReloading } from './utils'
export { Operation } from './model/operation'
export { NimbusJSX, setFragmentFactory } from './jsx'
export { If, Then, Else, ForEach, Switch } from './jsx/structure'
export * from './operations'
import { Component } from './model/component'
export { componentValidation, ValidationFn, ValidationNode } from './validation'
export { getGlobalState } from './global-state'

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace JSX {
    /**
     * A JSX element is equivalent to an instance of the class Component.
     */
    type Element = Component

    interface ElementChildrenAttribute {
      children: 'children',
    }

    interface IntrinsicElements {
      /**
       * `<component />` is used to reference a component in the frontend, but instead of using it directly when
       * building the screen, we advise creating a strictly typed functional component that is easier to use.
       *
       * @example
       * ```tsx
       * interface MyCustomTextProps {
       *   children: string,
       *   fontFamily: string,
       *   color: string,
       *   size: string,
       * }
       *
       * export const MyCustomText: FC<MyCustomTextProps> = ({ children, id, ...other }) => (
       *   <component
       *     name="my-custom-text"
       *     namespace="custom"
       *     id={id}
       *     properties={{ ...other, text: children }}
       *   />
       * )
       * ```
       *
       * Then, when building your screen:
       * ```tsx
       * const MyScreen = () => (
       *   <MyCustomText color="#000">Hello World</MyCustomText>
       * )
       * ```
       *
       * This JSX element is equivalent to the class {@link Component}. Read its documentation for details on each
       * attribute.
       *
       */
      component: Element,
    }
  }
}
