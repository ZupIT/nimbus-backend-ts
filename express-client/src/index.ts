/**
 * @module @zup-it/nimbus-backend-express
 */

export { NimbusApp } from './nimbus-app'
export { Screen, ScreenRequest } from './screen'
export { RouteMap } from './route'
export { isMobile, isWeb } from './utils/headers'
