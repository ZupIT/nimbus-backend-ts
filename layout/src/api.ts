/* This file is only used for exporting the API documentation. We do this so we can have a full documentation without
exporting types that doesn't make sense. */

/**
 * @module @zup-it/nimbus-backend-layout
 */

export * from './'
