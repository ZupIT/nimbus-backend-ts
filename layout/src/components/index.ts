export * from './box'
export * from './column'
export * from './flow-column'
export * from './flow-row'
export * from './image'
export * from './lazy-column'
export * from './lazy-row'
export * from './lifecycle'
export * from './positioned'
export * from './row'
export * from './screen'
export * from './scroll-view'
export * from './types/box'
export * from './types/container'
export * from './stack'
export * from './text'
export * from './touchable'
