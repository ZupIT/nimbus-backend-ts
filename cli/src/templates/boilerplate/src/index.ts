
import express from 'express'
import cors from 'cors'
import { NimbusApp } from '@zup-it/nimbus-backend-express'
import configs from '../nimbus-ts.json'
import { routes } from './screens'

const expressApp = express()
expressApp.use(cors()).use(express.json())
expressApp.listen(configs.port, () => {
  console.log(`App listening at http://localhost:${configs.port}`)
})

new NimbusApp(expressApp, routes, { basePath: configs.basePath })
