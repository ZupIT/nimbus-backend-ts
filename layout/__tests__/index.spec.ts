import { mapValues } from 'lodash'
import * as layout from 'src'

describe('Layout module', () => {
  it('should export', () => expect(mapValues(layout, v => typeof v)).toMatchSnapshot())
})
