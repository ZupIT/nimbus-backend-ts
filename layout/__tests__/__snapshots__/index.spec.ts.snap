// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout module should export 1`] = `
Object {
  "Box": "function",
  "Column": "function",
  "FlowColumn": "function",
  "FlowRow": "function",
  "Lifecycle": "function",
  "LocalImage": "function",
  "Positioned": "function",
  "RemoteImage": "function",
  "Row": "function",
  "ScreenComponent": "function",
  "ScrollView": "function",
  "Stack": "function",
  "Text": "function",
  "Touchable": "function",
}
`;
